emitter.go

```diff
@@ -55,13 +54,12 @@ func Sync(e *Event) { e.Flags = e.Flags | FlagSync }
 
 // New returns just created Emitter struct. Capacity argument
 // will be used to create channels with given capacity
-func New(capacity uint) *Emitter {
-	return &Emitter{
-		Cap:         capacity,
-		listeners:   make(map[string][]listener),
-		middlewares: make(map[string][]func(*Event)),
-		isInit:      true,
-	}
+// optionally include a matcher to use or nil to use the default
+// which implements path.Match() from the system library
+func New(capacity uint, matcher Matcher) *Emitter {
```

Comment:
How about to create another function/constructor, like:

```go
func NewWithMatcher(capacity uint, matcher Matcher) *Emitter {
```

or make it a variadic function, like:

```go
func New(capacity uint, matcher ...Matcher) *Emitter {
```
rather than introducing a broken change?

I would prefer to don't change the existing library's API if it's not a but fix.